sphinx/search/__init__.py

```diff
@@ -583,17 +582,18 @@ def get_js_stemmer_rawcode(self) -> str | None:
 
     def get_js_stemmer_code(self) -> str:
         """Returns JS code that will be inserted into language_data.js."""
-        if self.lang.js_stemmer_rawcode:
-            base_js_path = _NON_MINIFIED_JS_PATH / 'base-stemmer.js'
-            language_js_path = _NON_MINIFIED_JS_PATH / self.lang.js_stemmer_rawcode
-            base_js = base_js_path.read_text(encoding='utf-8')
-            language_js = language_js_path.read_text(encoding='utf-8')
-            return (
-                f'{base_js}\n{language_js}\nStemmer = {self.lang.language_name}Stemmer;'
-            )
-        else:
+        if not self.lang.js_stemmer_rawcode:
             return self.lang.js_stemmer_code
 
+        base_js_path = _MINIFIED_JS_PATH / 'base-stemmer.js'
+        language_js_path = _MINIFIED_JS_PATH / self.lang.js_stemmer_rawcode
+        return '\n'.join((
+            base_js_path.read_text(encoding='utf-8'),
+            language_js_path.read_text(encoding='utf-8'),
+            f'const Stemmer = {self.lang.language_name}Stemmer;',
+            f'globalThis.Stemmer = {self.lang.language_name}Stemmer;',
```

Comment:
What's the reasoning for providing `globalThis` access to the stemmer?

Reply:
Does it fix/improve anything relative to the existin sans-`globalThis` implementation?  I would hope that if/when Snowball begins using ESM modules, we could `export` directly and not need to pollute any global scope(s).

Reply:
My thought process is that it might not be great to provide the appearance of multi-environment (browser, node, ...) compatibility without any intention for the code to run (now or in future) in those environments -- even if, theoretically, using `globalThis` is a recommendable practice generally.

Reply:
I've changed to `window`.